File: src/Projects/Android/ChatSystem/ChatSystemServiceLib/src/main/java/org/csystem/app/chatsystem/service/mapper/IUserMapper.java
package org.csystem.app.chatsystem.service.mapper;

import org.csystem.app.chatsystem.repository.entity.User;
import org.csystem.app.chatsystem.service.dto.LoginInfosDTO;
import org.csystem.app.chatsystem.service.dto.UserDTO;
import org.csystem.app.chatsystem.service.dto.UserSaveDTO;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(implementationName = "UserMapperImpl", componentModel = "spring")
public interface IUserMapper {

    User toUser(UserSaveDTO userSaveDTO);

    @Mapping(target = "loginInfosDTOs", source = "loginInfosDTO")
    UserDTO toUserDTO(User user, LoginInfosDTO loginInfosDTO);
}

File: src/Projects/Android/PaymentApplication/DataServiceLib/src/main/java/org/csystem/android/app/payment/data/service/mapper/IUserMapper.kt
package org.csystem.android.app.payment.data.service.mapper

import org.csystem.android.app.payment.data.service.dto.UserSaveDTO
import org.csystem.android.app.payment.repository.entity.User
import org.mapstruct.Mapper

@Mapper(implementationName = "UserMapperImpl")
interface IUserMapper {
    fun toUser(userSaveDTO: UserSaveDTO) : User
    fun toUSerSaveDTO(user: User) : UserSaveDTO
}